include/boost/geometry/algorithms/detail/overlay/overlay.hpp

```diff
@@ -102,38 +102,42 @@ inline void get_ring_turn_info(TurnInfoMap& turn_info_map, Turns const& turns, C
             = target_operation == operation_union
             ? operation_intersection
             : operation_union;
+    static const bool is_union = target_operation == operation_union;
 
     for (auto const& turn : turns)
     {
         bool cluster_checked = false;
         bool has_blocked = false;
 
-        if (is_self_turn<OverlayType>(turn) && turn.discarded)
+        if (turn.discarded && (turn.method == method_start || is_self_turn<OverlayType>(turn)))
         {
-            // Discarded self-turns don't count as traversed
+            // Discarded self-turns or start turns don't need to block the ring
             continue;
         }
 
-        for (auto const& op : turn.operations)
+        for (int i = 0; i < 2; i++)
         {
+            auto const& op = turn.operations[i];
+            auto const& other_op = turn.operations[1 - i];
             ring_identifier const ring_id = ring_id_by_seg_id(op.seg_id);
 
-            if (! is_self_turn<OverlayType>(turn)
-                && (
-                    (BOOST_GEOMETRY_CONDITION(target_operation == operation_union)
-                      && op.enriched.count_left > 0)
-                  || (BOOST_GEOMETRY_CONDITION(target_operation == operation_intersection)
-                      && op.enriched.count_right <= 2)))
```

Comment:
I'm glad this condition could be removed. These counts can have issues with floating point precision, because they count open and close polygons on either sides, and open/close can be ambiguous for cases where in/out segments are very close.